src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -2940,13 +2940,15 @@ private DynamicMetaObject BinaryBitwiseAnd(DynamicMetaObject target, DynamicMeta
 
                 if (numericTarget != null && numericArg != null)
                 {
-                    var expr = exprGenerator(numericTarget.Expression.Cast(target.LimitType).Cast(opType),
-                        numericArg.Expression.Cast(numericArg.LimitType).Cast(opType)).Cast(typeof(object));
+                    var expr = exprGenerator(numericTarget.Expression.Cast(numericTarget.LimitType).Cast(opType),
+                                             numericArg.Expression.Cast(numericArg.LimitType).Cast(opType));
 
                     if (target.LimitType.GetTypeInfo().IsEnum)
                     {
-                        expr = expr.Cast(target.LimitType).Cast(typeof(object));
+                        expr = expr.Cast(target.LimitType);
                     }
+
+                    expr = expr.Cast(typeof(object));
```

Comment:
Why was this cast moved outside of the IsEnum code? None of the tests you wrote demonstrate the need here.